app.js

```diff
@@ -0,0 +1,129 @@
+import express from 'express';
+import bodyParser from 'body-parser';
+import crypto from 'crypto';
+import cors from 'cors';
+import { products, users, carts, orders } from './storage.js';
+
+
+const app = express();
+app.use(bodyParser.json());
+app.use(cors());
+
+// Валідація email та password
+const validateEmail = (email) => {
+    const emailRegex = /^[a-zA-Z0-9._-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,}$/;
+    return emailRegex.test(email);
+};
+
+const validatePassword = (password) => {
+    const passwordRegex = /^(?=.*[a-z])(?=.*[A-Z])(?=.*\d)(?=.*[@$!%*?&])[A-Za-z\d@$!%*?&]{8,}$/;
+    return passwordRegex.test(password);
+};
+
+
+// POST - реєстрація користувача
+app.post('/api/register', async (req, res) => {
+    const { email, name, password } = req.body;
+
+    if (!validateEmail(email)) {
+        return res.status(400).json({ error: 'Невалідний email' });
+    }
+    if (!validatePassword(password)) {
+        return res.status(400).json({ error: 'Пароль не відповідає вимогам' });
+    }
+
+    const id = crypto.randomUUID();
+    const newUser = {
+        id,
+        email,
+        name
+    };
+
+    users.push(newUser);
+
+    res.setHeader('x-user-id', id);
+    res.status(201).json(newUser);
+});
+
+// GET - отримати всі продукти
+app.get('/api/products', (req, res) => {
+    res.send(products);
+});
+
+// GET - отримати продукт за ID
+app.get('/api/products/:productId', async (req, res) => {
+    const product = products.find(p => p.id === parseInt(req.params.productId));
+    if (!product) {
+        return res.status(404).json({ error: 'Продукт не знайдено' });
+    }
+    res.json(product);
+});
+
+//  Перевірка заголовка x-user-id
+const authMiddleware = (req, res, next) => {
```

Comment:
I would suggest to create a separate file and move middleware there and export it